inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ModifierFeatureEditor.java

```diff
@@ -0,0 +1,554 @@
+import org.apache.wicket.behavior.AttributeAppender;
+import org.apache.wicket.feedback.IFeedback;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.form.ChoiceRenderer;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.markup.repeater.RefreshingView;
+import org.apache.wicket.markup.repeater.util.ModelIteratorAdapter;
+import org.apache.wicket.model.CompoundPropertyModel;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.model.PropertyModel;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.googlecode.wicket.kendo.ui.form.dropdown.DropDownList;
+import de.tudarmstadt.ukp.clarin.webanno.api.AnnotationSchemaService;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.action.AnnotationActionHandler;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.exception.AnnotationException;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.feature.editor.FeatureEditor;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.AnnotatorState;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.FeatureState;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.LinkWithRoleModel;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.util.WebAnnoCasUtil;
+import de.tudarmstadt.ukp.clarin.webanno.constraints.evaluator.PossibleValue;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationFeature;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationLayer;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.model.Tag;
+import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaAjaxFormComponentUpdatingBehavior;
+import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaModel;
+import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaModelAdapter;
+import de.tudarmstadt.ukp.inception.kb.KnowledgeBaseService;
+import de.tudarmstadt.ukp.inception.kb.graph.KBHandle;
+
+public class ModifierFeatureEditor
+    extends FeatureEditor
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ModifierFeatureEditor.class);
+
+    private static final long serialVersionUID = 7469241620229001983L;
+
+    private @SpringBean AnnotationSchemaService annotationService;
+
+    private @SpringBean KnowledgeBaseService kbService;
+    private @SpringBean FactLinkingService factService;
+
+    private WebMarkupContainer content;
+
+    // // For showing the status of Constraints rules kicking in.
+    // private RulesIndicator indicator = new RulesIndicator();
+
+    @SuppressWarnings("rawtypes")
+    private Component field;
+    private boolean hideUnconstraintFeature;
```

Comment:
We have a feature called "constraints" which can be used to change the choices available in the dropdown field of string features with tagsets. Theoretically, this feature could also be applied to concept linking dropdown, but for the moment it would be ok to remove related functionality from the concept dropdowns.